File: seed/ruby-sdk-v2/openapi-request-body-ref/lib/seed/team_member/types/update_team_member_request.rb
# frozen_string_literal: true

module Seed
  module TeamMember
    module Types
      class UpdateTeamMemberRequest < Internal::Types::Model
        field :team_member_id, -> { String }, optional: false, nullable: false
        field :given_name, -> { String }, optional: true, nullable: false
        field :family_name, -> { String }, optional: true, nullable: false
        field :email_address, -> { String }, optional: true, nullable: false
      end
    end
  end
end

File: seed/php-sdk/openapi-request-body-ref/src/TeamMember/Requests/UpdateTeamMemberRequest.php
<?php

namespace Seed\TeamMember\Requests;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class UpdateTeamMemberRequest extends JsonSerializableType
{
    /**
     * @var ?string $givenName
     */
    #[JsonProperty('given_name')]
    public ?string $givenName;

    /**
     * @var ?string $familyName
     */
    #[JsonProperty('family_name')]
    public ?string $familyName;

    /**
     * @var ?string $emailAddress
     */
    #[JsonProperty('email_address')]
    public ?string $emailAddress;

    /**
     * @param array{
     *   givenName?: ?string,
     *   familyName?: ?string,
     *   emailAddress?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->givenName = $values['givenName'] ?? null;
        $this->familyName = $values['familyName'] ?? null;
        $this->emailAddress = $values['emailAddress'] ?? null;
    }
}
